src/mod_pushoff_fcm.erl

```diff
@@ -0,0 +1,238 @@
+%%%
+%%% Copyright (C) 2015  Christian Ulrich
+%%% Copyright (C) 2016  Vlad Ki
+%%% Copyright (C) 2017  Dima Stolpakov =)
+%%%
+%%% This program is free software; you can redistribute it and/or
+%%% modify it under the terms of the GNU General Public License as
+%%% published by the Free Software Foundation; either version 2 of the
+%%% License, or (at your option) any later version.
+%%%
+%%% This program is distributed in the hope that it will be useful,
+%%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+%%% General Public License for more details.
+%%%
+%%% You should have received a copy of the GNU General Public License along
+%%% with this program; if not, write to the Free Software Foundation, Inc.,
+%%% 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+%%%
+
+%% This implements the "legacy" binary API
+
+-module(mod_pushoff_fcm).
+
+-author('christian@rechenwerk.net').
+-author('proger@wilab.org.ua').
+-author('dimskii123@gmail.com').
+
+-behaviour(gen_server).
+-compile(export_all).
+-export([init/1,
+         handle_info/2,
+         handle_call/3,
+         handle_cast/2,
+         terminate/2,
+         code_change/3]).
+
+-include("logger.hrl").
+
+-define(MAX_PAYLOAD_SIZE, 2048).
+-define(MESSAGE_EXPIRY_TIME, 86400).
+-define(MESSAGE_PRIORITY, 10).
+-define(MAX_PENDING_NOTIFICATIONS, 1).
```

Comment:
same here, why keep a separate send queue to send messages one-by-one and rely on httpc to do the work?